lib/krane/kubernetes_resource.rb

```diff
@@ -498,7 +511,7 @@ def validate_spec_with_kubectl(kubectl)
     end
 
     def validate_with_dry_run_option(kubectl, dry_run_option)
-      command = ["apply", "-f", file_path, dry_run_option, "--output=name"]
+      command = ["create", "-f", file_path, dry_run_option, "--output=name"]
```

Comment:
Unfortunately I think we need to switch these validation methods around a bit more, because `create` doesn't accept the `--server-dry-run` feature, at least as of 1.12 (that's a possible value of `dry_run_option` here). Adding a raw `Secret` using `generateName` to your new fixture should showcase the problem.